coderd/autobuild/executor/lifecycle_executor.go

```diff
@@ -268,37 +268,59 @@ func build(ctx context.Context, store database.Store, workspace database.Workspa
 		return xerrors.Errorf("Unsupported transition: %q", trans)
 	}
 
-	newProvisionerJob, err := store.InsertProvisionerJob(ctx, database.InsertProvisionerJobParams{
-		ID:             provisionerJobID,
-		CreatedAt:      now,
-		UpdatedAt:      now,
-		InitiatorID:    workspace.OwnerID,
-		OrganizationID: template.OrganizationID,
-		Provisioner:    template.Provisioner,
-		Type:           database.ProvisionerJobTypeWorkspaceBuild,
-		StorageMethod:  priorJob.StorageMethod,
-		FileID:         priorJob.FileID,
-		Tags:           priorJob.Tags,
-		Input:          input,
-	})
-	if err != nil {
-		return xerrors.Errorf("insert provisioner job: %w", err)
-	}
-	_, err = store.InsertWorkspaceBuild(ctx, database.InsertWorkspaceBuildParams{
-		ID:                workspaceBuildID,
-		CreatedAt:         now,
-		UpdatedAt:         now,
-		WorkspaceID:       workspace.ID,
-		TemplateVersionID: priorHistory.TemplateVersionID,
-		BuildNumber:       priorBuildNumber + 1,
-		ProvisionerState:  priorHistory.ProvisionerState,
-		InitiatorID:       workspace.OwnerID,
-		Transition:        trans,
-		JobID:             newProvisionerJob.ID,
-		Reason:            buildReason,
-	})
+	lastBuildParameters, err := store.GetWorkspaceBuildParameters(ctx, priorHistory.ID)
 	if err != nil {
-		return xerrors.Errorf("insert workspace build: %w", err)
+		return xerrors.Errorf("fetch prior workspace build parameters: %w", err)
 	}
-	return nil
+
+	return store.InTx(func(db database.Store) error {
```

Comment:
I'm not sure this is strictly necessary as each workspace is processed in its own transaction.

Reply:
I'm rather thinking about the case when it inserts a record to the workspace table, but it fails to insert parameters for some reason. In this case, it's better to rollback the entire transaction.